### ISSUE TITLE
Properties that start with `_` should be suppressed by `PropertyName`

### ISSUE THREAD

**Issue by Goooler** [pinterest/ktlint#2714]
## Expected Behavior

No errors are thrown.

## Observed Behavior

```sh
ktlint Foo.kt
/Users/goooler/StudioProjects/DemoApp/adapter/src/main/kotlin/io/goooler/demoapp/adapter/rv/core/Foo.kt:4:17: Backing property not allowed when 'private' modifier is missing (standard:backing-property-naming)
/Users/goooler/StudioProjects/DemoApp/adapter/src/main/kotlin/io/goooler/demoapp/adapter/rv/core/Foo.kt:4:17: Backing property is only allowed when a matching property or function exists (standard:backing-property-naming)

Summary error count (descending) by rule:
  standard:backing-property-naming: 2
```

## Steps to Reproduce

`Foo.kt`:

```kt
open class Foo {

  @Suppress("PropertyName")
  protected val _bar = mutableListOf<String>()
}
```

Run `ktlint Foo.kt`.

## Context

Related to:
- #2332
- #2612

## Your Environment

* Version of ktlint used: CLI 1.3.0

**Comment by paul-dingemans** [comment]
`@Suppress(PropertyName)` currently only suppresses the `property-naming` rule ([code](https://github.com/pinterest/ktlint/blob/32fd86f1d60a724f42affb2beafab055a2006478/ktlint-rule-engine/src/main/kotlin/com/pinterest/ktlint/rule/engine/internal/SuppressionLocatorBuilder.kt#L44)).
